client/js/socket-events/uploads.js

```diff
@@ -0,0 +1,15 @@
+"use strict";
+
+const $ = require("jquery");
+const socket = require("../socket");
+const wrapCursor = require("undate").wrapCursor;
+
+socket.on("upload:success", (url) => {
+	const fullURL = new URL(url, location);
+	const textbox = document.getElementById("input");
+	wrapCursor(textbox, fullURL, "");
+});
+
+socket.on("upload:failure", () => {
```

Comment:
This is never emitted.

Reply:
Leave it in for plugins?

Reply:
Let's worry about that later, remove it for now.

Reply:
done